test/performance/comm/low-level/many-to-one.chpl

```diff
@@ -49,9 +50,11 @@ proc main() {
   if printTimings then
     timer.start();
 
+  allLocalesBarrier.reset(numTasksPerNode);
+  begin { coforall 1..numTasksPerNode do allLocalesBarrier.barrier(); }
```

Comment:
Yeah, can do -- I debated doing that originally anyways. I always debating having locale 0 participate in the coforall and just doing a top-level barrier across the locales with the assumption that local task creation should take roughly the same amount of time anyways. The diff for that was:

```diff
index d40aa9141e..5369b93364 100644
--- a/test/performance/comm/low-level/many-to-one.chpl
+++ b/test/performance/comm/low-level/many-to-one.chpl
@@ -51,11 +51,11 @@ proc main() {
     timer.start();

-  allLocalesBarrier.reset(numTasksPerNode);
-  begin { coforall 1..numTasksPerNode do allLocalesBarrier.barrier(); }

-  coforall locIdx in 1..numWorkerNodes with ( ref x0 ) {
-    on Locales(locIdx) {
+  coforall locIdx in 0..numWorkerNodes with ( ref x0 ) do on Locales(locIdx) {

+    allLocalesBarrier.barrier();
+    if locIdx != 0 {
       var numOpsPerTask: [1..numTasksPerNode] int;
       var timePerTask: [1..numTasksPerNode] real;

@@ -66,8 +66,6 @@ proc main() {
         var t: Timer;
         var tElapsed: real;

-        allLocalesBarrier.barrier();
-
         t.start();

         while true {
```

Reply:
(so we could do something like that if you think it's cleaner)

Reply:
I debated coding that way originally just to make it more clear that this was a coforall-stmt across only the "worker" nodes rather than all the nodes as most loops over Locales are.  I finally decided that it was nice to avoid introducing one more level of logic (the if-stmt) inside the loop, and the leading `1` in the coforall's range was sufficient.  But with the change to a barrier in which all nodes must participate I think I feel the other way -- that it's better to include node 0 in the coforall across nodes and simplify the barrier coding.  I think I might write it like this, though:

```chapel
coforall loc in Locales ... {
  const locIdx = loc.id;
  allLocalesBarrier.barrier();
  if locIdx in 1..numWorkerNodes {
    ...
  }
}
```
(I've also found myself subsequently wondering if we'd get a more pure communication performance measurement, especially for GETs where we seem variable,  if the target node were not the one that had initiated the coforall across ndoes.  That is, if the target node were node `numLocales-1` rather than node `0`.  But I didn't investigate if that improved anything.)

Reply:
>(I've also found myself subsequently wondering if we'd get a more pure communication performance measurement, especially for GETs where we seem variable, if the target node were not the one that had initiated the coforall across ndoes. That is, if the target node were node numLocales-1 rather than node 0. But I didn't investigate if that improved anything.)

Yeah, that's an interesting thought. We do know that local 0 will be spin-waiting (and using 100% of a cpu) waiting on the end count.